bot/exts/moderation/infraction/_utils.py

```diff
@@ -298,3 +308,29 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+def cap_timeout_duration(duration: datetime.datetime | relativedelta) -> tuple[bool, datetime.datetime]:
+    """Caps the duration of a duration to Discord's limit."""
+    now = arrow.utcnow()
+    capped = False
+    if isinstance(duration, relativedelta):
+        duration += now
+
+    if duration > now + MAXIMUM_TIMEOUT_DAYS:
+        duration = now + MAXIMUM_TIMEOUT_DAYS - datetime.timedelta(minutes=1)  # Duration cap is exclusive.
+        capped = True
+    elif duration > now + MAXIMUM_TIMEOUT_DAYS - datetime.timedelta(minutes=1):
+        # Duration cap is exclusive. This is to still allow specifying "28d".
+        duration -= datetime.timedelta(minutes=1)
+    return capped, duration
+
+
+async def notify_timeout_cap(bot: Bot, ctx: Context, user: discord.Member) -> None:
+    """Notifies moderators about a timeout duration being capped."""
```

Comment:
Imperative mood is our prevailing style.
```suggestion
    """Notify moderators about a timeout duration being capped."""
```